java/runtime-native/src/main/java/org/ray/spi/StateStoreProxy.java

```diff
@@ -31,5 +31,19 @@
    * getAddressInfo.
    * @return list of address information
    */
-  List<AddressInfo> getAddressInfo(final String nodeIpAddress, int numRetries);
+  List<AddressInfo> getAddressInfo(final String nodeIpAddress, 
+                                   final String redisAddress,
+                                   int numRetries);
+
+  /**
+   *
+   * Get address info of one node from primary redis.
+   *
+   * @param nodeIpAddress Usually local ip address.
+   * @param redisAddress The primary redis address.
+   * @return A list of SchedulerInfo which contains raylet name and raylet port.
+   * @throws Exception No redis client exception.
+   */
+  List<AddressInfo> getAddressInfoHelper(final String nodeIpAddress,
```

Comment:
This method shouldn't be part of the interface, because it's the implementation detail. I think it'd make more sense to make it a abstract method in `BaseStateStoreProxy`. (Maybe also give it a better name.)

Reply:
Good idea.